packages/react-input/src/components/Input/useInputStyles.ts

```diff
@@ -1,28 +1,225 @@
+// TODO(sharing) should these be shared somewhere?
+const fieldHeights = {
+  small: '24px',
+  medium: '32px',
+  large: '40px',
+};
+const borderRadius = (theme: Theme) => theme.global.borderRadius.medium;
+const backgroundColors = {
+  /** for outline/filledLighter */
+  filledLighter: (theme: Theme) => theme.alias.color.neutral.neutralBackground1,
+  filledDarker: (theme: Theme) => theme.alias.color.neutral.neutralBackground3,
+  /** for underline */
+  transparent: (theme: Theme) => theme.alias.color.neutral.transparentBackground,
+};
+
+const useStyles = makeStyles({
+  root: theme => ({
+    display: 'flex',
+    alignItems: 'stretch',
+    minHeight: fieldHeights.medium, // minHeight not height in case of tall bookends
+    fontFamily: theme.global.type.fontFamilies.base,
+    ...contentSizes.body1(theme),
+    boxSizing: 'border-box',
+    '*, *:before, *:after': {
+      boxSizing: 'border-box',
+    },
+  }),
+  rootSmall: theme => ({
+    minHeight: fieldHeights.small,
+    ...contentSizes.caption1(theme),
+  }),
+  rootLarge: theme => ({
+    minHeight: fieldHeights.large,
+    ...contentSizes[400](theme),
+  }),
+  rootInline: {
+    display: 'inline-flex',
+  },
+  rootFilled: theme => ({
+    // optional shadow for filled appearances
+    boxShadow: theme.alias.shadow.shadow2,
+    borderRadius: borderRadius(theme), // needed for shadow
+  }),
+  rootDisabled: {
+    cursor: 'not-allowed',
+  },
+  input: theme => ({
+    flexGrow: 1,
+    border: 'none', // input itself never has a border (this is handled by inputWrapper)
+    // TODO(design) is this what "Left/Right Padding in content" means? why do we need it?
+    padding: `0 ${horizontalSpacing.xxs}`,
+    color: theme.alias.color.neutral.neutralForeground1,
+    // These are set on root or inputWrapper but don't inherit
+    ...contentSizes.body1(theme),
+    background: backgroundColors.filledLighter(theme), // for outline/filledLighter
+
+    '::placeholder': {
+      color: theme.alias.color.neutral.neutralForeground4, // for outline/underline/filledLighter
+    },
+    ':focus-visible': {
+      outline: 'none', // disable default browser outline
+    },
+  }),
+  inputSmall: theme => ({
+    ...contentSizes.caption1(theme),
+  }),
+  inputLarge: theme => ({
+    ...contentSizes[400](theme),
+    // TODO(design) is this what "Left/Right Padding in content" means? why do we need it?
+    padding: `0 ${horizontalSpacing.sNudge}`,
+  }),
+  inputFilledDarker: theme => ({
+    backgroundColor: backgroundColors.filledDarker(theme),
+  }),
+  inputUnderline: theme => ({
+    backgroundColor: backgroundColors.transparent(theme),
+  }),
+  inputDisabled: theme => ({
+    color: theme.alias.color.neutral.neutralForegroundDisabled,
+    backgroundColor: theme.alias.color.neutral.transparentBackground,
+    cursor: 'not-allowed',
+    '::placeholder': {
+      color: theme.alias.color.neutral.neutralForegroundDisabled,
+    },
+  }),
+  inputWrapper: theme => ({
+    flexGrow: 1,
```

Comment:
Do you want flex-wrap: no-wrap for the input wrapper to prevent the bookends moving to different lines than the input?

Reply:
Good point, thanks!